### PR TITLE
Change: Refactor maps

### PR DESCRIPTION
**Pull request by craiglonsdale**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "Mapping places and things with angular.",
   "main": "index.js",
```
Comment (jimmyhillis) [inline package.json]: Update the version to `2.0.0` cause we renamed things.

File: mapping.js [hunk 2]
```diff
@@ -240,113 +223,121 @@ function ($scope, $timeout, $q, $window, GoogleService, MarkerService) {
...
-        },
-        link: function ($scope, element, attr) {
-            $scope.$parent.default_icon = $scope.markerIcon;
-        }
+      restrict: 'A',
+      scope: {
+        markerIcon: '=markerIcon'
+      },
+      link: function ($scope) {
+        $scope.$parent.defaultIcon = $scope.markerIcon;
```
Comment (jimmyhillis) [inline mapping.js]: This should pull in the controller `GoogleMapController` and then do controllerAs so we don't need the scope yeah?

Comment (jimmyhillis) [inline mapping.js]: Actually this is a service now so it needs to bring in `GoogleMapService` and do `GoogleMapService.defaultIcon` or whatever

File: mapping.js [hunk 3]
```diff
@@ -240,113 +223,121 @@ function ($scope, $timeout, $q, $window, GoogleService, MarkerService) {
...
-        }
+            function rezoomMap() {
+              GoogleMapService.options.zoom = parseInt(attr.zoom, 10);
+              GoogleMapService.googlemap.setZoom(GoogleMapService.options.zoom);
+            }
+            attr.$observe('latitude', repositionMap);
+            attr.$observe('longitude', repositionMap);
+            attr.$observe('zoom', rezoomMap);
+            GoogleMapService.refresh();
+            $rootScope.$broadcast('google_map_initialized', GoogleMapService.refresh);
```
Comment (jimmyhillis) [inline mapping.js]: This isn't even used. So random. This can be removed.

File: mapping.js [hunk 4]
```diff
@@ -240,113 +223,121 @@ function ($scope, $timeout, $q, $window, GoogleService, MarkerService) {
...
+   * @name mapping.controller:GoogleMapController
+   * @description
+   * Service interactive Google Map and interactive map markers
+   */
+  // @ngInject
+  .controller('GoogleMapController', function ($scope, MarkerService, GoogleMapService) {
     // Update markers
-    $scope.$watch(function () { return MarkerService.markers(); }, function () {
-        $scope.refresh();
+    $scope.$watch(function () {
```
Comment (jimmyhillis) [inline mapping.js]: `$scope.$watch(MarkerService.getMarkers, GoogleMapService.refresh)` which is much nicer!

File: mapping.js [hunk 5]
```diff
@@ -240,113 +223,121 @@ function ($scope, $timeout, $q, $window, GoogleService, MarkerService) {
...
+      const newAnchor = angular.extend({ x: newSize.width / 2, y: newSize.height }, anchor);
+      return {
+        anchor: new google.maps.Point(newAnchor.x, newAnchor.y),
+        origin: new google.maps.Point(0, 0),
+        scaledSize: new google.maps.Size(newScaledSize.width, newScaledSize.height),
+        size: new google.maps.Size(newSize.width, newSize.height),
+        url: url
+      };
     };
+  })
```
Comment (jimmyhillis) [inline mapping.js]: kevin

File: mapping.js [hunk 6]
```diff
@@ -168,48 +151,48 @@ function ($scope, $timeout, $q, $window, GoogleService, MarkerService) {
...
+        service.markers = [];
+        angular.forEach(markers, function (marker) {
+          if (marker.icon) {
+            marker.icon = service.iconFromURL(marker.icon.url,
+                                             marker.icon.size,
+                                             marker.icon.scaledSize,
+                                             marker.icon.anchor);
+          }
+          service.addMarker(new google.maps.LatLng(marker.latitude, marker.longitude),
+                               marker.title,
```
Comment (jimmyhillis) [inline mapping.js]: formatttting

File: package.json (hunk 1 above)
Comment (craiglonsdale) [inline package.json]: lol

File: mapping.js (hunk 5 above)
Comment (craiglonsdale) [inline mapping.js]: bacon?

**Pull request merged**